samples/posix/eventfd/Makefile.host

```diff
@@ -1,4 +1,4 @@
 # This makefile builds the sample for a POSIX system, like Linux
 
-eventfd: src/main.c
-	$(CC) $^ -o $@
+eventfd: samples/posix/eventfd/src/main.c
```

Comment:
> Updated Makefile.hostfile for samples to be used with make from 'zephyr' directory, similar to west build command.

I find this change somewhat unnecessary, it can't be compared with the `west build` command because `west` works anywhere (either in the `zephyr` folder or in the `zephyr/samples/posix/eventfd` folder). Changing the path in this Makefile doesn't fix its limitation, it's just a matter of preference (i.e. I usually build in the `zephyrproject` folder, this change wouldn't benefit me) and it deviates from other existing `Makefile.host`s in the `net/socket` examples.

Also, before this change, a developer can copy the sample to anywhere and `make` would work as-is, after this change, the sample must be in the relative path for the Makefile to work.

TL;DR: This change only increases the limitation of the Makefile